### PR TITLE
que3.5

### PR DESCRIPTION
**Pull request by ashok-choudhary-0**

### CODE REVIEW

File: controllers/userController.js [hunk 1]
```diff
@@ -156,21 +157,39 @@ const deleteUserAddresses = async (req, res) => {
 
 const forgotPassword = async (req, res) => {
   const { username } = req.body
+  const transporter = nodemailer.createTransport({
```
Comment (amrit56789) [inline controllers/userController.js]: add this method after jwt token,

File: controllers/userController.js [hunk 2]
```diff
@@ -156,21 +157,39 @@ const deleteUserAddresses = async (req, res) => {
...
   try {
     const userData = await userModel.findOne({ where: { username } })
     if (userData) {
       const passwordResetToken = jwt.sign({
         data: `${userData.username} ${userData.password}`
       }, process.env.jwtSecKey, { expiresIn: 60 * 10 });
       await userModel.update({ passwordResetToken: passwordResetToken }, { where: { username } })
-      res.status(200).send(passwordResetToken)
 
+      const info = {
```
Comment (amrit56789) [inline controllers/userController.js]: what info?

File: controllers/userController.js [hunk 3]
```diff
@@ -156,21 +157,39 @@ const deleteUserAddresses = async (req, res) => {
...
+        from: '"ashok.kumar@innotechteam.in',
```
Comment (amrit56789) [inline controllers/userController.js]: don't pass directly take from env file.

File: controllers/userController.js [hunk 4]
```diff
@@ -156,21 +157,39 @@ const deleteUserAddresses = async (req, res) => {
...
+        to: "ashok03012000@gmail.com",
```
Comment (amrit56789) [inline controllers/userController.js]: this one also

File: controllers/userController.js [hunk 5]
```diff
@@ -156,21 +157,39 @@ const deleteUserAddresses = async (req, res) => {
...
+        subject: "conformation for password reset",
+        html: `<b>Password reset url:->  http://localhost:8000/user/forgot-password/${passwordResetToken}</b>`,
+      }
+      transporter.sendMail(info, (err, result) => {
+        if (err) {
+          res.status(500).send(err)
+        } else {
+          res.status(200).send({ message: "mail sent successfully", info })
```
Comment (amrit56789) [inline controllers/userController.js]: First character should be capital. why are you passing small letter everywhere.

Comment (amrit56789) [inline controllers/userController.js]: send info.response

File: controllers/userController.js [hunk 6]
```diff
@@ -156,21 +157,49 @@ const deleteUserAddresses = async (req, res) => {
...
   try {
     const userData = await userModel.findOne({ where: { username } })
     if (userData) {
       const passwordResetToken = jwt.sign({
         data: `${userData.username} ${userData.password}`
       }, process.env.jwtSecKey, { expiresIn: 60 * 10 });
       await userModel.update({ passwordResetToken: passwordResetToken }, { where: { username } })
-      res.status(200).send(passwordResetToken)
 
+      const transporter = nodemailer.createTransport({
```
Comment (yuvrajInno) [inline controllers/userController.js]: move this code to a separate file, which will be use to do only email related stuff. @ashok-choudhary-0

File: controllers/userController.js [hunk 7]
```diff
@@ -156,21 +157,49 @@ const deleteUserAddresses = async (req, res) => {
...
+        service: "gmail",
+        auth: {
+          user: process.env.nodemailerEmail,
+          pass: process.env.nodemailerEmailPassword
+        }
+      })
+      const emailInformation = {
+        from: process.env.senderMail,
+        to: process.env.receiverMail,
```
Comment (yuvrajInno) [inline controllers/userController.js]: receiverMail should be user's email @ashok-choudhary-0

**Review by yuvrajInno — CHANGES_REQUESTED** [review]

**Pull request merged**